crypto/evp/evp_enc.c

```diff
@@ -336,7 +360,20 @@ int EVP_EncryptUpdate(EVP_CIPHER_CTX *ctx, unsigned char *out, int *outl,
             return 0;
         }
     }
+
+    /* Past this point inl has to be in bytes. */
     i = ctx->buf_len;
+    /*
+     * Aliasing in and out in this way is counter intuitive and only supported
+     * for legacy applications.
+     * Don't rely on this to stay the same in the next major version.
+     */
+    if (!is_pointer_offset_by_bias(in, out, i)
+            && is_any_overlapping(out, inl + (bl - 1), in, inl)) {
```

Comment:
I'm not exactly happy that there are actual subroutine calls. As low-level gets faster, overhead becomes more significant (on short inputs), and it's only appropriate working toward reducing it. For reference, as soon as subroutine is non-static, compiler will refuse to inline it if you compile with -fPIC, and we do so by default.

Reply:
I could make the three is_xxx static, and move them . plus this ugly PTRDIFF_T into the header
file.
Do you know why we can not just use uintptr_t ?

Reply:
> Do you know why we can not just use uintptr_t ?

This is a little bit wrong question. It's not about *present* can or cannot, but rather about *past* choices, from time when there were compilers in support matrix that didn't have it. However! Even today it's not that obvious that it would be appropriate to use it, because it's claimed *optional* by standard. Well, one can as well choose to take the same path as with other \*_t types, i.e. provide own typedefs in case they are missing. See e_os2.h to get the idea. And it looks like one is likely to do something like that... If we choose to take this path that is...